src/helpers/semester.ts

```diff
@@ -0,0 +1,94 @@
+const isSpringSemester = (date: Date) => {
+  const month = date.getMonth();
+  return month > 1 && month < 8;
+}
+
+export class SemesterInfo {
+  readonly springSemester: boolean;
+  readonly year: number;
+
+  constructor(springSemester: boolean, year: number) {
+    this.springSemester = springSemester;
+    this.year = year;
+  }
+
+  static now() {
+    const now = new Date();
+    return new SemesterInfo(isSpringSemester(now), now.getFullYear());
+  }
+
+  static parse(text: string) {
+    if (text.length !== 4) return null;
+
+    const prefix = text.substring(0, 2);
+    const suffix = text.substring(2);
+
+    const parsedSuffix = parseInt(suffix);
+    if (isNaN(parsedSuffix)) return null;
+
+    const year = parsedSuffix + 2000;
+
+    switch (prefix) {
+      case "FS":
+        return new SemesterInfo(true, year);
+      case "HS":
+        return new SemesterInfo(false, year);
+      default:
+        return null;
+    }
+  }
+
+  toString() {
+    const prefix = this.springSemester ? "FS" : "HS";
+    const suffix = this.year.toString().substring(2);
+
+    return prefix + suffix;
+  }
+
+  addSemesters(count: number) {
```

Comment:
I really dislike the naming `addSemesters` and `removeSemester`...
Could we maybe get a static `increaseSemesters(count: number, baseSemester: SemesterInfo)` and static `decreaseSemesters(count: number, baseSemester: SemesterInfo)`? Small change, but feels much cleaner.

Reply:
I called it `plus` and `minus` now.